CMSgov/mint-app
# [EASI-3676] Update risk questions (backend)

# EASI-3676

## Changes and Description

- GQL Schema
- SQL storage methods
- SQL Flyway migration
- Go DTO model



## How to test this change

- Disable flyway migration
- Bring up new containers
- Insert data using provided SQL (see below)
- Validate data in db editor of your choice
- Enable flyway migration
- Run scripts/dev db:migrate
- Validate data in db editor of your choice
- Validate unit tests are stable

```
INSERT INTO "public"."user_account"("id", "username", "is_euaid", "common_name", "locale", "email", "given_name", "family_name", "zone_info", "has_logged_in") VALUES('762d3239-8dca-4486-8a7b-fe6255eab1e8', 'MINT', 'TRUE', 'MINT Doe', 'en_US', 'MINT@local.cms.gov', 'MINT', 'Doe', 'America/Los_Angeles', 'TRUE');


/* YES + TWO_SIDED */
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by")
    VALUES('0940d64f-17bc-4f33-ad7c-c82a08f1b190', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');
INSERT INTO "public"."plan_participants_and_providers"("id", "model_plan_id", "participant_assume_risk", "risk_type", "created_dts", "created_by")
    VALUES('a904f2d6-3d26-4202-b67e-328186257000', '0940d64f-17bc-4f33-ad7c-c82a08f1b190', 'true', 'TWO_SIDED', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');

/* YES + ONE_SIDED */
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by")
    VALUES('0940d64f-17bc-4f33-ad7c-c82a08f1b191', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');
INSERT INTO "public"."plan_participants_and_providers"("id", "model_plan_id", "participant_assume_risk", "risk_type", "created_dts", "created_by")
    VALUES('a904f2d6-3d26-4202-b67e-328186257001', '0940d64f-17bc-4f33-ad7c-c82a08f1b191', 'true', 'ONE_SIDED', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');

/* YES + CAPITATION */
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by")
    VALUES('0940d64f-17bc-4f33-ad7c-c82a08f1b192', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');
INSERT INTO "public"."plan_participants_and_providers"("id", "model_plan_id", "participant_assume_risk", "risk_type", "created_dts", "created_by")
    VALUES('a904f2d6-3d26-4202-b67e-328186257002', '0940d64f-17bc-4f33-ad7c-c82a08f1b192', 'true', 'CAPITATION', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');

/* YES + OTHER */
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by")
    VALUES('0940d64f-17bc-4f33-ad7c-c82a08f1b193', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');
INSERT INTO "public"."plan_participants_and_providers"("id", "model_plan_id", "participant_assume_risk", "risk_type", "created_dts", "created_by")
    VALUES('a904f2d6-3d26-4202-b67e-328186257003', '0940d64f-17bc-4f33-ad7c-c82a08f1b193', 'true', 'OTHER', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');

/* NO */
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by")
    VALUES('0940d64f-17bc-4f33-ad7c-c82a08f1b194', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');
INSERT INTO "public"."plan_participants_and_providers"("id", "model_plan_id", "participant_assume_risk", "created_dts", "created_by")
    VALUES('a904f2d6-3d26-4202-b67e-328186257004', '0940d64f-17bc-4f33-ad7c-c82a08f1b194', 'false', '2023-12-19 15:32:57.692596+00', '762d3239-8dca-4486-8a7b-fe6255eab1e8');
```

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
